crates/matrix-sdk/src/encryption/backups/mod.rs

```diff
@@ -62,9 +65,15 @@ use crate::{
 #[derive(Debug, Clone)]
 pub struct Backups {
     pub(super) client: Client,
+    cached_exists_on_server: Arc<RwLock<Option<bool>>>,
```

Comment:
The `Backups` object is an API object. As such, it might not be long-lived. Does it live long-enough, in the caller's code, to actually make use of that cached value? Otherwise, we'd likely need to put that in the `ClientInner` itself, or as a subfield.

Other point: instead of an `Arc<RwLock<Option<>>>`, we could likely use a `OnceCell`, a cell location that's initialized at most once and forever? tokio provides one that can be initialized with async code, and it also has `try_init` that can handle `Result`. Not sure if that can work with the clearing that happens in `clear_exists_on_server_cache`…

Reply:
Darnit, you are right - the `Backups` object lives very briefly. Back to the drawing board...

Reply:
Fixed in 5d5ac13f38

Thanks for spotting this!